diff --git a/src/game/Chat.cpp b/src/game/Chat.cpp
diff --git a/src/game/SpellMgr.cpp b/src/game/SpellMgr.cpp
diff --git a/src/game/Unit.cpp b/src/game/Unit.cpp
index 6757b12..5dbd686 100644
--- a/src/game/Unit.cpp
+++ b/src/game/Unit.cpp
@@ -8716,6 +8716,22 @@ uint32 Unit::SpellDamageBonus(Unit *pVictim, SpellEntry const *spellProto, uint3
             }
         }
     }
+	
+	AuraList const& mDummy = owner->GetAurasByType(SPELL_AURA_DUMMY);
+    for(AuraList::const_iterator i = mDummy.begin(); i != mDummy.end(); ++i)
+    {
+        SpellEntry const *spell = (*i)->GetSpellProto();
+        //Fire and Brimstone
+        if (spell->SpellFamilyName == SPELLFAMILY_WARLOCK && spell->SpellIconID == 3173)
+        {
+            if (pVictim->HasAuraState(AURA_STATE_CONFLAGRATE) && (spellProto->SpellFamilyName == SPELLFAMILY_WARLOCK && spellProto->SpellFamilyFlags & UI64LIT(0x0002004000000000)))
+            {
+                DoneTotalMod *= ((*i)->GetModifier()->m_amount+100.0f) / 100.0f;
+                break;
+            }
+        }
+    }
+
 
     // Custom scripted damage
     switch(spellProto->SpellFamilyName)
diff --git a/src/game/Unit.h b/src/game/Unit.h
