diff --git a/src/game/Unit.cpp b/src/game/Unit.cpp
index 6757b12..197ed80 100644
--- a/src/game/Unit.cpp
+++ b/src/game/Unit.cpp
@@ -7440,6 +7440,15 @@ bool Unit::HandleProcTriggerSpell(Unit *pVictim, uint32 damage, Aura* triggeredB
                 trigger_spell_id = 50475;
                 basepoints[0] = damage * triggerAmount / 100;
             }
+			 // Glyph of Death Grip
+            else if (auraSpellInfo->Id == 58628)
+            {
+                // remove cooldown of Death Grip
+                if (GetTypeId() == TYPEID_PLAYER)
+                    ((Player*)this)->RemoveSpellCooldown(82);
+                return true;
+            }
+
             // Blade Barrier
             else if (auraSpellInfo->SpellIconID == 85)
             {
