diff --git a/src/game/Unit.cpp b/src/game/Unit.cpp
index 6757b12..e5214fe 100644
--- a/src/game/Unit.cpp
+++ b/src/game/Unit.cpp
@@ -4924,6 +4924,12 @@ bool Unit::HandleDummyAuraProc(Unit *pVictim, uint32 damage, Aura* triggeredByAu
                     triggered_spell_id = 26654;
                     break;
                 }
+				// Glyph of Blocking
+            if (dummySpell->Id == 58375)
+            {
+                triggered_spell_id = 58374;
+                break;
+			}
                 // Twisted Reflection (boss spell)
                 case 21063:
                     triggered_spell_id = 21064;
