diff --git a/src/game/Unit.cpp b/src/game/Unit.cpp
index 6757b12..bd777ee 100644
--- a/src/game/Unit.cpp
+++ b/src/game/Unit.cpp
@@ -6684,6 +6684,13 @@ bool Unit::HandleDummyAuraProc(Unit *pVictim, uint32 damage, Aura* triggeredByAu
                 triggered_spell_id = 61607;
                 break;
             }
+			// Unholy Blight
+            if (dummySpell->Id == 49194)
+            {
+                basepoints0 = triggerAmount * damage / 100;
+                triggered_spell_id = 50536;
+                break;
+            }
             // Vendetta
             if (dummySpell->SpellFamilyFlags & UI64LIT(0x0000000000010000))
             {
