diff --git a/src/game/SpellAuras.cpp b/src/game/SpellAuras.cpp
index 840ef36..5e751b3 100644
--- a/src/game/SpellAuras.cpp
+++ b/src/game/SpellAuras.cpp
@@ -5155,6 +5155,7 @@ void Aura::HandleAuraModIncreaseHealth(bool apply, bool Real)
         case 44055: case 55915: case 55917: case 67596:     // Tremendous Fortitude (Battlemaster's Alacrity)
         case 50322:                                         // Survival Instincts
         case 54443:                                         // Demonic Empowerment (Voidwalker)
+		case 55233:                                         // Vampiric Blood
         {
             if(Real)
             {
@@ -5164,7 +5165,8 @@ void Aura::HandleAuraModIncreaseHealth(bool apply, bool Real)
                     // recalculate to full amount at apply for proper remove
                     if (GetId() == 54443)
                         m_modifier.m_amount = m_target->GetMaxHealth() * m_modifier.m_amount / 100;
-
+					if (GetId()==55233)
+                        m_modifier.m_amount *=  m_target->GetMaxHealth() / 100;
                     m_target->HandleStatModifier(UNIT_MOD_HEALTH, TOTAL_VALUE, float(m_modifier.m_amount), apply);
                     m_target->ModifyHealth(m_modifier.m_amount);
                 }
